### PR TITLE
Support automatic krel fast forward

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

We now add an additional `--non-interactive` flag to `krel ff`, which avoids requiring user interaction. If no `--branch` is provided, then `krel ff` will no longer fail, but try to determine the latest release branch. If this release branch is found, it checks if a fast forward is required by testing if the latest final tag (like v1.23.0) exists on the remote. If the tag exists, krel will will do nothing. Otherwise it will do the fast forward.

#### Which issue(s) this PR fixes:

Refers to #2386

#### Special notes for your reviewer:

/hold

Requires https://github.com/kubernetes-sigs/release-sdk/pull/18 and https://github.com/kubernetes/release/pull/2389

#### Does this PR introduce a user-facing change?

```release-note
- Added `--non-interactive` flag to `krel ff`,
- Automatically determine the `krel ff --branch` if not provided.
  If the branch is found, `krel ff` will check if a fast forward is required or not by
  testing the availability of the latest final tag (like v1.23.0).
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
/hold cancel
Rebased

**Comment by saschagrunert** [comment]
```
403 API rate limit of 60 still exceeded until 2022-01-14 12:16:23 +0000 UTC, not making remote request.
```

File: pkg/fastforward/fastforward.go [hunk 1]
```diff
@@ -80,18 +76,39 @@ func (f *FastForward) Run() error {
...
+		}
+
+		logrus.Info("Checking if branch is available on the default remote")
+		branchExists, err := f.RepoHasRemoteBranch(repo, branch)
+		if err != nil {
+			return errors.Wrap(err, "checking if branch exists on the default remote")
+		}
+		if !branchExists {
+			return errors.New("branch does not exist on the default remote")
+		}
```
Comment (puerco) [inline pkg/fastforward/fastforward.go:111]: I'm curious here, why is there a different check logic when specifying the branch? I get it that we will be running it mostly in the latest branch and this completes the other case.
But are you thinking of a possible use for ff'ing other branches now?

Comment (saschagrunert) [reply pkg/fastforward/fastforward.go:111]: In theory the tool could be used to sync any branch combination. I’d wanted to preserve that while only adding automation for our additional use case.

I think we can use the manual path by ff-ing this cycle again, if we communicate it early enough (in the next week).

The automation can be used to verify that it works and could go live in 1.25. WDYT?

**Pull request merged**